mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala

```diff
@@ -92,6 +92,56 @@ class RankingMetrics[T: ClassTag](predictionAndLabels: RDD[(Array[T], Array[T])]
     }.mean()
   }
 
+  /**
+   * Returns the mean average precision (MAP) at ranking position k of all the queries.
```

Comment:
It's kind of "average precision up to position k" rather than at? but I understand its conventional to call this average precision at k. Is there anything we can link to to further explain?

Reply:
I can give the reference to some literature, hope this would give user a better idea.